src/Voron/GlobalFlushingBehavior.cs

```diff
@@ -66,29 +68,26 @@ private void VoronEnvironmentFlushing()
                 while (true)
                 {
                     avoidDuplicates.Clear();
-                    var maybeNeedSync = _maybeNeedToSync.Count;
-                    var millisecondsTimeout = 15000 - maybeNeedSync;
-                    if (millisecondsTimeout <= 0 ||
-                        _flushWriterEvent.Wait(millisecondsTimeout) == false)
+                    
+                    if (_flushWriterEvent.Wait(5000) == false)
```

Comment:
In my checks, I ran a patch on 28.66 KB documents that each iteration multiple the document by 100
There were transactions that took for up to 7-8 seconds and lead to `SyncEnvironments(force: true);` even the system was in pressure and didn't get to journals threshold.
Maybe we should raise the time before `SyncEnvironments(force: true);`  ?

Reply:
Probably we should have better detection of "no activity". Maybe we could use `StorageEnvironment.LastWorkTime`, also check if there is no active write transaction right now